### PR TITLE
Fix date issues.

### LINKED ISSUE #102: [PE-D] Future Date Flaw in Add Expenditure
Allow future date in the add expenditure date parameter. I think you should block any date entry beyond today's date

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: lihao-InfoSec/ped#1

### LINKED ISSUE #104: [PE-D] Wrong Error Message Shown for Invalid Date Entry
The expected error message should be asking the user to input an valid date instead.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: lihao-InfoSec/ped#2

### LINKED ISSUE #110: [PE-D] Valid Date Bug
I believe 29/02/2020 is an valid date but why is it got rejected. Same goes to 30/03/2020. I believe this application does not take in any year before 2021.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: lihao-InfoSec/ped#3

### PR DESCRIPTION
**Pull request by limjierui**
Fixes #110 
Fixes #104 
Fixes #102 

Fix date issues.
Add more Junit test for Add and Update Command

### CODE REVIEW

**Review by SeanHoWB — COMMENTED** [review]
Looks good, remember to update UG to reflect the changes

File: src/main/java/seedu/mindmymoney/command/AddCommand.java [hunk 1]
```diff
@@ -147,6 +150,44 @@ public void addExpenditure() throws MindMyMoneyException {
         System.out.print(System.lineSeparator());
     }
 
+    /**
+     * Checks if parsed date is after the current date.
+     *
+     * @param date date that is parsed in.
+     * @throws MindMyMoneyException throws an exception when the date parsed is after current date.
+     */
+    public static void checkAfterCurrentDate(LocalDate date) throws MindMyMoneyException {
```
Comment (GlendonNotGlen) [inline src/main/java/seedu/mindmymoney/command/AddCommand.java:159]: Maybe can consider making this function a boolean instead?

**Review by GlendonNotGlen — COMMENTED** [review]
Overall a great implementation and will be great for adding robustness in our program!

File: src/main/java/seedu/mindmymoney/command/AddCommand.java (hunk 1 above)
Comment (limjierui) [reply src/main/java/seedu/mindmymoney/command/AddCommand.java:159]: Hmm, I did that previously but I realised there will be alot of if statements, hence I decided to just throw the exception in the function itself.

**Pull request merged**